src/data_write.py

```diff
@@ -8,175 +8,213 @@
-    # "noise_in_raw_band"         : None, # Average noise in the sampling band (input sample rate) (TODO units??)
-    # "rx_bandwidth"              : None, # if the noise_in_raw_band is provided, the rx_bandwidth should be provided!
-    "num_samps"                 : None, # Number of samples in the sampling period.
-    "antenna_arrays_order"      : None, # States what order the data is in. Describes the data
-                                        # layout.
-    "data_descriptors"          : None, # Denotes what each data dimension represents.
-    "data_dimensions"           : None, # The dimensions in which to reshape the data.
-    "data_normalization_factor" : None, # The scale of all of the filters, multiplied, for a total
-                                        # scaling factor to normalize by.
-    "data"                      : [],   # A contiguous set of samples (complex float) at given
-                                        # sample rate
-    "correlation_descriptors"   : None, # Denotes what each acf/xcf dimension represents.
-    "correlation_dimensions"    : None, # The dimensions in which to reshape the acf/xcf data.
-    "averaging_method"          : None, # A string describing the averaging method, ex. mean, median
-    "scheduling_mode"           : None, # A string describing the type of scheduling time at the
-                                        # time of this dataset.
-    "main_acfs"                 : [],   # Main array autocorrelations
-    "intf_acfs"                 : [],   # Interferometer array autocorrelations
-    "xcfs"                      : [],   # Crosscorrelations between main and interferometer arrays
-    "gps_locked"                : None, # Boolean True if the GPS was locked during the entire
-                                        # integration period
-    "gps_to_system_time_diff"   : None, # Max time diff in seconds between GPS and system/NTP time
-                                        # during the integration period.
-    "agc_status_word"           : None, # 32 bits, a '1' in bit position corresponds to an AGC
-                                        # fault on that transmitter
-    "lp_status_word"            : None  # 32 bits, a '1' in bit position corresponds to a low power
-                                        # condition on that transmitter
-}
-
-TX_TEMPLATE = {
-    "tx_rate"               : [],
-    "tx_center_freq"        : [],
-    "pulse_timing_us"       : [],
-    "pulse_sample_start"    : [],
-    "tx_samples"            : [],
-    "dm_rate"               : [],
-    "decimated_tx_samples"  : [],
-}
+@dataclass(init=False)
+class SliceData:
+    """
+    This class defines all fields that need to be written by any type of data file. The 'groups' metadata lists
+    the applicable file types for each field.
+    """
+    agc_status_word: int = field(
+        metadata={'groups': ['antennas_iq', 'bfiq', 'rawacf', 'rawrf']})
```

Comment:
I'm not opposed to it. I thought that it looked a little cleaner with metadata on the second line because all of the 'groups' lists were aligned so it would be easy to read.